src/wallet/api.rs

```diff
@@ -1223,10 +1260,11 @@ impl Wallet {
 
         match select_coin(&output_groups, &coin_selection_option) {
             Ok(selection) => {
+                // Convert selected group indices back to individual UTXOs
                 let selected_utxos: Vec<(ListUnspentResultEntry, UTXOSpendInfo)> = selection
                     .selected_inputs
                     .iter()
-                    .map(|&index| unspents[index].clone())
+                    .flat_map(|&group_index| grouped_utxos[group_index].clone())
```

Comment:
All the logic in the match statement should remain intact. We already have all the utxos in `unspent`, no need to get them again from `grouped_utxos`